### DISCUSSION TITLE
How do I disable the automatic require adding?

### DISCUSSION THREAD

**Discussion by TuggyNE** [BetterThanTomorrow/calva#2135] [Q&A]
Somewhere in the stack of clj-kondo, clojure-lsp, and Calva is an apparently undocumented feature (#1663, perhaps?) that automatically adds `require`s to the `ns` when you accept an autocomplete suggestion for an ns alias that it recognizes. This is great, except that it's usually wrong, so I want to turn it off. How?

Details of wrongness, which could go in a bug report once I figure out which project it goes in:
* It can't detect that I'm in a rich comment with extra `(require)` forms that already have that ns as that alias, so it cheerfully adds duplicates (outside the comment, where they are not useful).
* It's incapable of recognizing ns prefix groups, such as
  ```clojure
  (:require
    (com.example.shared-prefix
      [foo :as foo :refer [bar baz]]
      [quux :as quux :refer [corge grault]]))
  ```
  So if I type `foo/waldo` it will add a duplicate as `[com.example.shared-prefix.foo :as foo]`. I use these only for company-specific namespaces, but wouldn't you know it, there are a lot of those.
* Occasionally it will interfere with var naming; I have to remember that I can't use autocomplete as a text shortcut for longer names, but only _to actually complete full symbols_, which is an annoying difference. If I used the feature for its intended purpose more, this would be an acceptable tradeoff, but I usually add requires preemptively when I know I need them, so the majority of uses are unintended.
* Undoing wipes out the text _and_ the ns addition. So each misfire creates at least as much work as each success saves.

**Comment by PEZ** [comment]
We should have a look at #1663 and see if Calva can do this better. I don't know how to disable the feature. You mean clojure-lsp settings do not have a switch for it?

**Reply by TuggyNE** [reply]
I don't know exactly which project is doing it (although apparently it isn't Calva), but I looked through https://clojure-lsp.io/settings/ a while ago and didn't see anything relevant. The closest is `:auto-add-ns-to-new-files?`, but that's only for blank files.

**Reply by PEZ** [reply]
Calva runs and manages the clojure-lsp server so it can be debated if it is Calva or not. Let's ask @ericdallo if there's a way to disable it.

**Reply by ericdallo** [reply]
clojure-lsp doesn't have a switch for it, feel free to open a issue there for that
